app/assets/javascripts/lib/ranger_events_dispatcher.js

```diff
@@ -0,0 +1,33 @@
+var RangerWebSocket = function(url){
+    var conn = new WebSocket(url);
+
+    var callbacks = {};
+
+    this.bind = function(event_name, callback){
+        callbacks[event_name] = callbacks[event_name] || [];
+        callbacks[event_name].push(callback);
+        return this;// chainable
+    };
+
+    this.send = function(payload){
+        conn.send( payload ); // <= send JSON data to socket server
+        return this;
+    };
+
+    // dispatch to the right handlers
+    conn.onmessage = function(evt){
```

Comment:
After sending jwt you will receive a message like `{"success": {"message": "Authorized[...]"}}` or `{"error":{"message":"Authentication failed"}}`, need to handle them properly.